spec/models/supporting_evidence_attachment_spec.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe SupportingEvidenceAttachment, type: :model do
+  describe '#obscured_filename' do
+    context 'for a filename longer than five characters' do
+      context 'for a filename containing letters' do
+        let(:attachment) do
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'MySecretFile.pdf' }.to_json
+          )
+        end
+
+        it 'obscures all but the first 3 and last 2 characters of the filename' do
+          expect(attachment.obscured_filename).to eq('MyS*******le.pdf')
+        end
+      end
+
+      context 'for a filename containing numbers' do
+        let(:attachment) do
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'MySecretFile123.pdf' }.to_json
+          )
+        end
+
+        it 'obscures the filename' do
+          expect(attachment.obscured_filename).to eq('MyS**********23.pdf')
+        end
+      end
+
+      context 'for a filename containing underscores' do
+        let(:attachment) do
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'MySecretFile_123.pdf' }.to_json
+          )
+        end
+
+        it 'does not obscure the underscores' do
+          expect(attachment.obscured_filename).to eq('MyS*********_*23.pdf')
+        end
+      end
+
+      context 'for a filename containing dashes' do
+        let(:attachment) do
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'MySecretFile-123.pdf' }.to_json
+          )
+        end
+
+        it 'does not obscure the dashes' do
+          expect(attachment.obscured_filename).to eq('MyS*********-*23.pdf')
+        end
+      end
+
+      # Ensure Regexp still handles the file extension properly
+      context 'for a filename containing a dot' do
+        let(:attachment) do
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'MySecret.File123.pdf' }.to_json
+          )
+        end
+
+        it 'obsucres the filename properly' do
+          expect(attachment.obscured_filename).to eq('MyS*****.*****23.pdf')
+        end
+      end
+
+      context 'for a filename containing whitespace' do
+        let(:attachment) do
+          build(
+            :supporting_evidence_attachment,
+            file_data: { filename: 'My Secret File.pdf' }.to_json
+          )
+        end
+
+        it 'preserves the whitespace' do
+          expect(attachment.obscured_filename).to eq('My ****** **le.pdf')
+        end
+      end
+    end
```

Comment:
Most of these can be combined by checking that any random non-alphanumeric character (except a period) is not obscured, not only underscores and dashes.

Reply:
Not necessarily, if someone tweaks the regex they could miss one of these use cases, and I wanted to make sure the common demarcations were always supported in regression testing.